### ISSUE TITLE
target elements of xlink ill-defined

### ISSUE THREAD

**Issue by mikedo** [MPEGGroup/DASHSchema#59]
Many MPD elements support @xlink. Yet only the Period element is enabled to standalone in an "MPD". 
However the target of an xlink is not an MPD, but "fragment" elements which permit @xlink. DASH clients when faced with, e.g. Period-Replacement.mpd might assume it is a complete MPD per the file extension, but is really just an MPD fragment.
This is confusing for test asset management also (files with the .mpd extension not really MPDs).
@xlink MPD fragments really should be in their own namespace and not called (via filename or otherwise) an MPD.
The top level Period element added in the 4th Ed should be removed from the MPD schema as a standalone element; and it and all other elements that allow @xlink should be defined in a new schema, e.g. DASH-XLINK.xsd.

**Comment by paulhiggs** [comment]
Generally, it is not necessary for a XML-fragment to pass validation in a standalone manner, only once "included" in another XML document. This is because some items, such as xs:ID and xs:IDREF cannot have their value space validated until read into a complete instance document.

Any example files could use a different file type, i.e. `.mpd-fragment` to discern these from an XML document that complies with an MPD. It is also useful to note that XML fragments that are "xlinked into" another XML document instance often do not exist persistently, but are service side created ny some PHP or other scripting system.

**Comment by mikedo** [comment]
The spec text is vague at best and I believe we need a precise specification about what can be returned in an xlink resolution.  I think calling the fragments *.mpd will create great confusion and significant interoperability issues.  I believe it is safe to say that existing DASH clients will not handle this well.  We can define a filename extension however we like, e.g. .mpd-framgent, but we need a vocabulary definition to conform form to.  It is trivial to formally create a new xlink response document, which is entirely undefined today. It's just an import of DASH-MPD and a list of elements, which would also clarify the cardinality of the MPD elements (enforce 1-N elements).

**Comment by paulhiggs** [comment]
Only XML documents that validate against the DASH MPD schema should use the .MPD file extension. A new file extension i.e. `.mpd-fragment` with its own schema (which references types from the DASH MPD schema) seems like a good approach.

**Comment by mikedo** [comment]
Agreed.  But in the 4th Ed, we permit Period by itself as a valid "MPD".

**Comment by paulhiggs** [comment]
This is not reflected in the 4th Edition FDIS [w18609](http://wg11.sc29.org/doc_end_user/documents/127_Gothenburg/wg11/w18609.zip). Which takes precedence.

**Comment by mikedo** [comment]
Yes. This will need supporting text somewhere, perhaps AMD1.

**Comment by paulhiggs** [comment]
The FDIS does not have the root level period element, so why don't we revert the change in the electronic attachments and deal with t entirely in AMD1

**Comment by mikedo** [comment]
The 4th Ed schema is what it is at this point. Period as a stand-alone element is supported by the xlink spec text (although arguably in an incomplete manner). I think it would be better to address this all at once and not remove Period without a resolution for what xlink targets *should* be.

**Comment by jpiesing** [comment]
See also the discussion in http://mpegx.int-evry.fr/software/MPEG/DASH/spec/issues/11 about using xlink URLs with a fragment identifier to be more precise about what is returned in an xlink resolution.

**Comment by mikedo** [comment]
Thanks. A good discussion. But unfortunately that issue is MPEG-private and is now closed and deferred.

**Comment by paulhiggs** [comment]
Per previous discussions, we seem to be trying to apply MPD validation rules to fragments of an MPD (such as those used in xlink). II would probably not implement a xlink period resolver at a `.mpd` url, more likely the xlink URL would be `https://my-server.net/gimme-a-period?number=9` which is perfectly valid for `@xlink:href` in a `<Period>` element

**Comment by mikedo** [comment]
Right.  And AMD1 explicitly removes Period as a top level element.  I'll file an issue and remove it when I figure out what all changed in the latest FDAM1...

**Comment by paulhiggs** [comment]
Closing this as we won't do schema based validation of MPF fragments
